emlog.c

```diff
@@ -84,16 +84,20 @@
 static struct emlog_info *emlog_info_list = NULL;
 static bool emlog_autofree;
 static bool emlog_debug;
-
+static int emlog_max_size = 1024;
 static dev_t emlog_dev_type = 0;
-#define EMLOG_MINOR_BASE    1
-#define EMLOG_MINOR_COUNT   EMLOG_MAX_SIZE
+
 static struct cdev *emlog_cdev = NULL;
 static struct class *emlog_class = NULL;
 static struct device *emlog_dev_reg;
 
 module_param(emlog_autofree, bool, 0644);
 module_param(emlog_debug, bool, 0644);
+module_param(emlog_max_size, int, 0644);
+
+#define EMLOG_MINOR_BASE    1
+#define EMLOG_MINOR_COUNT   emlog_max_size
```

Comment:
I'm afraid, that this may confuse people.
When I see a macro like `EMLOG_MINOR_COUNT`, I expect a compile-time constant.
However, it turns out to be a global variable in this case.

It'd be nice if you could simply replace all `EMLOG_MINOR_COUNT` occurrences with `emlog_max_size`,
and add a comment to `emlog_init()` to explain why `emlog_max_size` is passed as a `count` argument to `alloc_chrdev_region()`.